x-pack/plugins/maps/public/plugin.ts

```diff
@@ -67,6 +73,12 @@ export interface MapsPluginSetupDependencies {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface MapsPluginStartDependencies {}
 
+declare module '../../../../src/plugins/ui_actions/public' {
+  export interface ActionContextMapping {
+    [ACTION_VISUALIZE_GEO_FIELD]: VisualizeFieldContext;
+  }
+}
```

Comment:
Do really every consumer of a trigger that wants to provide an action, needs to declare the context like this? It feels like this should somehow be possible in one individual place? cc @stacey-gammon

Reply:
It should be possible to only declare in a single place - the place where the trigger is being registered. Every producer of a trigger needs to declare context like this.  

That being said, I know this pattern is going away in favor of accessing typed triggers directly from the producer plugin, not off the generic registry. I'm not sure where the team is with that effort though, or if it's on a roadmap.  cc @streamich @Dosant

Reply:
@stratoula could we then so far move that type declaraction into the plugin where it's defined?